crates/bevy_reflect/src/fields.rs

```diff
@@ -24,20 +28,32 @@ impl NamedField {
     pub fn type_name(&self) -> &str {
         self.type_name.borrow()
     }
+
+    /// The `TypeId` of the field
+    pub fn type_id(&self) -> TypeId {
+        self.type_id
+    }
+
+    /// Check if the given type matches this field's type
+    pub fn is<T: Reflect>(&self) -> bool {
+        TypeId::of::<T>() == self.type_id
+    }
```

Comment:
A lot of info structs also have a helper function like this. Should this stay? Alternatively, it can be moved to a trait with a default implementation (this would probably include the `type_id()` method as well).